airflow/api_fastapi/core_api/routes/public/event_logs.py

```diff
@@ -49,3 +59,75 @@ async def get_event_log(
         event_log,
         from_attributes=True,
     )
+
+
+@event_logs_router.get("/", responses=create_openapi_http_exception_doc([401, 403]))
+async def get_event_logs(
+    limit: QueryLimit,
+    offset: QueryOffset,
+    session: Annotated[Session, Depends(get_session)],
+    order_by: Annotated[
+        SortParam,
+        Depends(
+            SortParam(
+                [
+                    "id",  # event_log_id
+                    "dttm",  # when
+                    "dag_id",
+                    "task_id",
+                    "run_id",
+                    "event",
+                    "execution_date",  # logical_date
+                    "owner",
+                    "extra",
+                ],
+                Log,
+            ).dynamic_depends()
+        ),
+    ],
+    dag_id: str | None = None,
+    task_id: str | None = None,
+    run_id: str | None = None,
+    map_index: int | None = None,
+    try_number: int | None = None,
+    owner: str | None = None,
+    event: str | None = None,
+    excluded_events: list[str] | None = Query(None),
+    included_events: list[str] | None = Query(None),
+    before: datetime | None = None,
+    after: datetime | None = None,
+) -> EventLogCollectionResponse:
+    """Get all Event Logs."""
+    base_select = select(Log).group_by(Log.id)
+    event_logs_select, total_entries = paginated_select(
+        base_select,
+        [
+            FilterParam(Log.dag_id, dag_id),
+            FilterParam(Log.task_id, task_id),
+            FilterParam(Log.run_id, run_id),
+            FilterParam(Log.map_index, map_index),
+            FilterParam(Log.event, event),
+            FilterParam(Log.try_number, try_number),
+            FilterParam(Log.owner, owner),
+            FilterParam(Log.event, excluded_events, "not_in"),
```

Comment:
Maybe we can create a concrete enum instead of a literal to avoid having string duplicated all over the place for `not_in`, etc.